server/api/api/endpoints/nuclio.py

```diff
@@ -47,3 +49,37 @@ async def list_api_gateways(
         ),
         auth_info,
     )
+
+
+@router.post("/projects/{project}/nuclio/api-gateways/{gateway}")
+async def create_api_gateway(
+    project: str,
+    gateway: str,
+    functions: list = Query(alias="functions"),
```

Comment:
Assuming we keep these params and not move to an object schema, the alias should be `function` since the REST API URL will look something like:
```
POST /projects/{project}/nuclio/api-gateways/{gateway}?function=<func1>&function=<func2>...
```
And fastapi will construct these query params into a list.
Anyway, I'm not sure why the `functions` have to be a query param at all.

Reply:
@theSaarco reworked and passing APIGateway entity as you suggested above :)